src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,940 @@
+            # }
+
+            for deposit in crypto_deposits:
+                self.__logger.debug("Transfer: %s", json.dumps(deposit))
+                self._process_transfer(deposit, intra_transactions)
+
+            # If user made more than 1000 transactions in a 90 day period we need to shrink the window.
+            if len(crypto_deposits) < 1000:
+                current_start = current_end + 1
+                current_end = current_start + _NINETY_DAYS_IN_MS
+            else:
+                # Binance sends latest record first ([0])
+                # CCXT sorts by timestamp, so latest record is last ([999])
+                current_start = int(crypto_deposits[999][_TIMESTAMP]) + 1  # times are inclusive
+                current_end = current_start + _NINETY_DAYS_IN_MS
+
+        # Process actual fiat deposits (no limit on the date range)
+        # Fiat deposits can also be pulled via CCXT fetch_deposits by cycling through legal_money
+        # Using the underlying api endpoint is faster for Binance.
+        # Note that this is the same endpoint as withdrawls, but with _TRANSACTIONTYPE set to 0 (for deposits)
+        fiat_deposits = self.client.sapiGetFiatOrders(params=({_TRANSACTIONTYPE: 0, _STARTTIME: self.start_time_ms, _ENDTIME: now_time}))
+        #    {
+        #      "code": "000000",
+        #      "message": "success",
+        #      "data": [
+        #        {
+        #          "orderNo": "25ced37075c1470ba8939d0df2316e23",
+        #          "fiatCurrency": "EUR",
+        #          "indicatedAmount": "15.00",
+        #          "amount": "15.00",
+        #          "totalFee": "0.00",
+        #          "method": "card",
+        #          "status": "Failed",
+        #          "createTime": 1627501026000,
+        #          "updateTime": 1627501027000
+        #        }
+        #      ],
+        #      "total": 1,
+        #      "success": True
+        #    }
+        if _DATA in fiat_deposits:
+            for deposit in fiat_deposits[_DATA]:
+                self.__logger.debug("Deposit: %s", json.dumps(deposit))
+                if deposit[_STATUS] == "Completed":
+                    self._process_deposit(deposit, in_transactions)
+
+    def _process_gains(self, in_transactions: List[InTransaction]) -> None:
+
+        ### Regular Dividends from Staking (including Eth staking) and Savings (Lending)
+
+        # We need milliseconds for Binance
+        current_start = self.start_time_ms
+        now_time = int(datetime.now().timestamp()) * 1000
+
+        # We will pull in 30 day periods. This allows for 16 assets with daily dividends.
+        current_end = current_start + _THIRTY_DAYS_IN_MS
+
+        while current_start < now_time:
+
+            # CCXT doesn't have a standard way to pull income, we must use the underlying API endpoint
+            dividends = self.client.sapiGetAssetAssetDividend(params=({_STARTTIME: current_start, _ENDTIME: current_end, _LIMIT: 500}))

```

Comment:
Is 500 arbitrary or is it a maximum limit defined in the docs? Use a global constant?